### ISSUE TITLE
Question of viewer

### ISSUE THREAD

**Issue by Sun-Jing-Kang** [YihangChen-ee/HAC#6]
How to view the trained model? Thanks!

I found the Scaffold-GS-viewer can't be used, the origin GS-viewer view like full black scene.

**Comment by YihangChen-ee** [comment]
Hi Sun-Jing-Kang,

The base structure of HAC follows Scaffold-GS, so it supports Scaffold-GS-viewer.
You should firstly decode the bitstreams to get the original model weight (ie, the `.ply` file), and input it into the Scaffold-GS-viewer.
Also, maybe you should firstly remove model weights related to context models before inputting the `.ply` file into the viewer.

:)

**Comment by YihangChen-ee** [comment]
Hi Sun-Jing-Kang,

HAC follows the base structure of Scaffol-GS, but there some minor modifications in the 3D model, which leads to the failure of viewing using the Scaffold-GS-viewer.

You can have a read on how we decode the model and render the testing images using the decoded model, and modify the Scaffold-GS-viewer or our HAC model to fit each other.

We will consider to directly support the Scaffold-GS-viewer in the future.

:)

**Comment by arcman7** [comment]
Does this HAC project support real time rendering?

**Comment by YihangChen-ee** [comment]
Yes. HAC essentially renders images under the 3DGS pipeline, thus is able to achieve real-time rendering. Please refer to the paragraph "Inference time" in our paper.

**Comment by arcman7** [comment]
Cool. If I am understanding you correctly, these bitstreams can be rendered by any rasterization method that is faithful the to the 3DGS rasterization implementation? Meaning SIBR or something similar to it can be used to render the output bitstreams from HAC?

**Comment by YihangChen-ee** [comment]
Hi, the bitstreams are not directly used for rendering though. Bitstreams are used to decode the 3DGS model (more precisely, Scaffold-GS model in our method), ie, the locations and the attributes. **_Note this decoding process only need to be run once to obtain the 3DGS model before the rendering process._** After the model is decoded and obtained, the rendering process (ie, the rasterization process) is totally the same to the Scaffold-GS, which is real-time.

**Comment by arcman7** [comment]
Oh, very nice! I did not realize that. I'm excited to try this out :)

**Comment by smart4654154** [comment]
> Yes. HAC essentially renders images under the 3DGS pipeline, thus is able to achieve real-time rendering. Please refer to the paragraph **Inference time** in our paper.

Is there a real-time viewing tool available now? Similar to SIBR in 3DGS

**Comment by YihangChen-ee** [comment]
> > Yes. HAC essentially renders images under the 3DGS pipeline, thus is able to achieve real-time rendering. Please refer to the paragraph **Inference time** in our paper.
> 
> Is there a real-time viewing tool available now? Similar to SIBR in 3DGS

You could use the [real-time viewing tool  of Scaffold-GS](https://github.com/city-super/Scaffold-GS/tree/main/SIBR_viewers) since we build the HAC framework based on Scaffold-GS. There are some minimal differences though, please remember to change the attribute **_scale_** to the version before activation in the decoded weights. (You could refer to the variable name **_decoded_version_** in our code to find these differences.)
